sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridSerializer.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using Azure.Core;
+
+namespace Azure.Messaging.EventGrid
+{
+    internal class EventGridSerializer : IUtf8JsonSerializable
+    {
+        public object _egEvent;
+        public CancellationToken _cancellationToken;
+
+        public EventGridSerializer(object egEvent, CancellationToken cancellationToken)
+        {
+            _egEvent = egEvent;
+            _cancellationToken = cancellationToken;
+        }
+        public void Write(Utf8JsonWriter writer)
+        {
+            JsonObjectSerializer serializer = new JsonObjectSerializer();
```

Comment:
Should we pull this out into an optional parameter of the service method and take an ObjectSerializer? If it is null, we would still create the JsonObjectSerializer, but this way customers can configure the serializer however they want and potentially even use other their own serializer implementation.